File: 01_Brute_Force/Level4/1339.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
vector<string> word;
vector<int> alpha(26);
int result = 0;

// 각 알파벳 별로 다 합쳤을때 얼마나 기여하는지 계산
void solve() {
	for (int i = 0; i < word.size(); i++) {
		int pow = 1;
		for (int j = word[i].size() - 1; j >= 0; j--) {
			int val = word[i][j] - 'A';
			alpha[val] += pow;
			pow *= 10;
		}
	}
    // 기여도가 높은 내림차순으로 정렬하여 9부터 대입시켜준 합을 구함.
	sort(alpha.begin(), alpha.end(), greater<int>());
	int num = 9;
	for (int i = 0; i < 26; i++) {
		if (alpha[i] == 0) break;
		result += (alpha[i] * num);
		num--;
	}
}

int main() {
	int num;
	string s;
	cin >> num;
	for (int i = 0; i < num; i++) {
		cin >> s;
		word.push_back(s);
	}
	solve();
	cout << result << endl;
	return 0;
}

File: 01_Brute_Force/Level4/1339.java
// Copyright@2023 Jihoon Lucas Kim <jihoon.lucas.kim@gmail.com>
import java.util.Scanner;
import java.util.Arrays;

public class Main {
	static String[] words;
	static int[] alpha = new int[26];
	static int result = 0;
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		words = new String[N];
		
		for(int i = 0; i < N; i++) {
			words[i] = sc.next();
		}
		
		solve();
		
		System.out.println(result);
	}
	
	// 각 알파벳 별로 다 합쳤을때 얼마나 기여하는지 계산
	static void solve() {
		
		for (int i = 0; i < words.length; i++) {
			int pow = 1;
			for (int j = words[i].length() - 1; j >= 0; j--) {
				int val = words[i].charAt(j) - 'A';
				alpha[val] += pow;
				pow *= 10;
			}
		}
		
		Arrays.sort(alpha);
		
	    // 기여도가 높은 순으로 9부터 대입시켜준 합을 구함.
		int num = 9;
		for (int i = 25; i >= 0; i--) {
			if (alpha[i] == 0) continue;
			result += (alpha[i] * num);
			num--;
		}
	}
}
